### PR TITLE
Add a UI test for section projects view in teacher dashboard

### PR DESCRIPTION
**Pull request by davidsbailey**
Also adds an optional `?sectionProjects=1` url param which the UI test uses to force section projects to be enabled (they are disabled by default in the test environment). Also fixes the links to the projects, which have been broken since we started showing the project list from pegasus.

### CODE REVIEW

File: pegasus/sites.v3/code.org/public/teacher-dashboard/index.haml [hunk 1]
```diff
@@ -99,6 +94,13 @@ title: <%= I18n.t('dashboard_landing_title').inspect %>
     return {
       templateUrl: 'teacher-dashboard/nav',
       controller: function($scope) {
+        // Show projects tab in the section nav if any of the following are true:
+        // 1. CDO.section_projects is set
+        // 2. DCDO.section_projects is set
+        // 3. sectionProjects url param is present
+        // 4. the projects tab is specified in the url
+        // The projects tab is accessible by direct url regardless of settings 1-3 above.
+        $scope.showSectionProjectsTab = #{show_section_projects} || $scope.tab === 'projects';
```
Comment (Bjvanminnen) [inline pegasus/sites.v3/code.org/public/teacher-dashboard/index.haml]: I'm not sure how interspersed these sorts of things are in our angular code, but if we're potentially going to refactor this one day, I wonder if it makes sense to try and keep our JS as pure as we can, and have all embedded ruby in one place, i.e. do something like

```
// at the beginning of the file
var showSectionProjects = #{show_section_projects};
...
// now here
$scope.showSectionProjectsTab = showSectionProjects || $scope.tab === 'projects';
``

File: pegasus/sites.v3/code.org/public/teacher-dashboard/index.haml [hunk 2]
```diff
@@ -589,34 +591,32 @@ title: <%= I18n.t('dashboard_landing_title').inspect %>
...
-        );
-      });
-
-      // the ng-select in the nav compares by reference not by value, so we can't just set
-      // selectedSection to section, we have to find it in sections.
-      $scope.sections.$promise.then(
-        function (sections) {
-          $scope.selectedSection = $.grep(sections, function(section) { return (section.id == $routeParams.id);})[0];
+  app.controller('SectionProjectsController', ['$scope', '$routeParams', 'sectionsService',
+                                             function($scope, $routeParams,  sectionsService) {
```
Comment (Bjvanminnen) [inline pegasus/sites.v3/code.org/public/teacher-dashboard/index.haml]: nit: i believe this should only be indented four spaces

File: pegasus/sites.v3/code.org/public/teacher-dashboard/index.haml (hunk 1 above)
Comment (Bjvanminnen) [reply pegasus/sites.v3/code.org/public/teacher-dashboard/index.haml]: Oh, i see now it was that way before. Any reason you changed it?

Comment (davidsbailey) [reply pegasus/sites.v3/code.org/public/teacher-dashboard/index.haml]: Nope, this change was incidental. Good point. I've moved the ruby to the beginning of the file.

File: pegasus/sites.v3/code.org/public/teacher-dashboard/index.haml [hunk 3]
```diff
@@ -7,8 +7,7 @@ title: <%= I18n.t('dashboard_landing_title').inspect %>
 - forbidden! unless dashboard_user[:user_type] == 'teacher'
 - error_string_none_selected = I18n.t('dashboard_error_none_selected').gsub(/"/,"\\\\\"")
 - error_string_other_section = I18n.t('dashboard_error_other_section').gsub(/"/,"\\\\\"")
-- show_section_projects = DCDO.get('section_projects', CDO.section_projects)
-
+- show_section_projects = DCDO.get('section_projects', CDO.section_projects) || !!request.params['sectionProjects']
```
Comment (Bjvanminnen) [inline pegasus/sites.v3/code.org/public/teacher-dashboard/index.haml]: `sectionProjects` query param is only used by UI tests, right? wonder if it's worth a comment here mentioning that explicitly.

File: pegasus/sites.v3/code.org/public/teacher-dashboard/index.haml [hunk 4]
```diff
@@ -589,34 +592,32 @@ title: <%= I18n.t('dashboard_landing_title').inspect %>
...
       );
-    }]);
-  }
+    });
+
+    // the ng-select in the nav compares by reference not by value, so we can't just set
+    // selectedSection to section, we have to find it in sections.
+    $scope.sections.$promise.then(
+      function (sections) {
+        $scope.selectedSection = $.grep(sections, function(section) { return (section.id == $routeParams.id);})[0];
```
Comment (Bjvanminnen) [inline pegasus/sites.v3/code.org/public/teacher-dashboard/index.haml]: nit: i believe our style guide says this function should be on multiple lines. also, is the == vs. === intentional?

Comment (davidsbailey) [reply pegasus/sites.v3/code.org/public/teacher-dashboard/index.haml]: Not intentional, just copied from existing code with bad style. I've fixed this here.

**Pull request merged**